cart/domain/cart/paymentselection.go

```diff
@@ -87,6 +96,34 @@ func NewSimplePaymentSelection(gateway string, method string, pricedItems Priced
 	return selection
 }
 
+//NewPaymentSelectionWithGiftCard - returns Selection with given giftcard charge type taken into account
+func NewPaymentSelectionWithGiftCard(gateway string, method string, pricedItems PricedItems, appliedGiftCards []AppliedGiftCard) (PaymentSelection, error) {
```

Comment:
Maybe we should refactor this to only expose one NewDefaultPaymentSelection(gateway,method,cart) PaymentSelection, error

That makes the public interface easier to use and understand